services/src/contexts/schema.sql

```diff
@@ -257,8 +257,245 @@ CREATE TYPE "ResultDescriptor" AS (
     plot "PlotResultDescriptor"
 );
 
--- seperate table for projects used in foreign key constraints
+CREATE TYPE "MockDatasetDataSourceLoadingInfo" AS (
+    points "Coordinate2D" []
+);
+
+CREATE TYPE "DateTimeParseFormat" AS (
+    fmt text,
+    has_tz boolean,
+    has_time boolean
+);
+
+CREATE TYPE "OgrSourceTimeFormatCustom" AS (
+    custom_format "DateTimeParseFormat"
+);
+
+CREATE TYPE "UnixTimeStampType" AS ENUM (
+    'EpochSeconds',
+    'EpochMilliseconds'
+);
+
+CREATE TYPE "OgrSourceTimeFormatUnixTimeStamp" AS (
+    timestamp_type "UnixTimeStampType",
+    fmt "DateTimeParseFormat"
+);
+
+CREATE TYPE "OgrSourceTimeFormat" AS (
+    -- oneOf
+    -- Auto
+    custom "OgrSourceTimeFormatCustom",
+    unix_time_stamp "OgrSourceTimeFormatUnixTimeStamp"
+);
+
+CREATE TYPE "OgrSourceDurationSpec" AS (
+    -- oneOf
+    infinite boolean, -- void
+    zero boolean, -- void
+    "value" "TimeStep"
+);
+
+CREATE TYPE "OgrSourceDatasetTimeTypeStart" AS (
+    start_field text,
+    start_format "OgrSourceTimeFormat",
+    duration "OgrSourceDurationSpec"
+);
+
+CREATE TYPE "OgrSourceDatasetTimeTypeStartEnd" AS (
+    start_field text,
+    start_format "OgrSourceTimeFormat",
+    end_field text,
+    end_format "OgrSourceTimeFormat"
+);
+
+CREATE TYPE "OgrSourceDatasetTimeTypeStartDuration" AS (
+    start_field text,
+    start_format "OgrSourceTimeFormat",
+    duration_field text
+);
+
+CREATE TYPE "OgrSourceDatasetTimeType" AS (
+    -- oneOf
+    -- None
+    "start" "OgrSourceDatasetTimeTypeStart",
+    start_end "OgrSourceDatasetTimeTypeStartEnd",
+    start_duration "OgrSourceDatasetTimeTypeStartDuration"
+);
+
+CREATE TYPE "CsvHeader" AS ENUM (
+    'Yes',
+    'No',
+    'Auto'
+);
+
+CREATE TYPE "FormatSpecificsCsv" AS (
+    header "CsvHeader"
+);
+
+CREATE TYPE "FormatSpecifics" AS (
+    -- oneOf
+    csv "FormatSpecificsCsv"
+);
+
+CREATE TYPE "OgrSourceColumnSpec" AS (
+    format_specifics "FormatSpecifics",
+    x text,
+    y text,
+    "int" text [],
+    "float" text [],
+    "text" text [],
+    bool text [],
+    "datetime" text [],
+    rename "TextTextKeyValue" []
+);
+
+CREATE TYPE "OgrSourceErrorSpec" AS ENUM (
+    'Ignore',
+    'Abort'
+);
+
+-- closed path is a polygon without stored bbox
```

Comment:
what is a closed path and how is that comment relevant here?

Reply:
You could wonder why not to use the `polygon` type.

Reply:
i still don't get it. 

Do you mean we define our own polygon type instead of the built-in polygon type because we do not want to store the bbox?

How is it ensured that the path is actually closed?

Reply:
[7b2f1cd](https://github.com/geo-engine/geoengine/pull/851/commits/7b2f1cd03f520a47b46dca63e9c7d7f3971bfc1d)

Our polygon's have closed paths in our rings so it corresponds 1:1.